### DISCUSSION TITLE
Stylesheet can't be loaded properly in production mode.

### DISCUSSION THREAD

**Discussion by kassihassan** [vanilophp/framework#154] [Q&A]
Hi everyone!
I discovered vanilo few months ago and I fall in love instantly. It was the kind of library that my Laravel projects were missing for a long time, so many thanks to the team behind it.
However, after my first deployment, my app failed loading the file https://software.kassihassan.com/css/appshell.css with the error message : 

1. The stylesheet https://software.kassihassan.com/css/appshell.css was not loaded because its MIME type, “text/html”, is not “text/css”. [order](https://software.kassihassan.com/admin/order)
2.Uncaught SyntaxError: expected expression, got '<'

I did some digging and found many results from people experienced the same problem, but none applies to my situation,  
I started building on top of this demo [https://github.com/vanilophp/demo](https://github.com/vanilophp/demo) because I didn't want to write the admin panel from scratch.
My hypothesis are:

- Laravel treats css/appshell.css as request to an endpoint of its routes, thus not found error. (I did encounter the same problem with images before, and I solved it by adding a specific route/controller for serving different kind of images)
- Demo used laraveI mix and bootstrap 4 to build the production assets, and I used on top of that tailwindcss with a mix plugin to build more storefront assets (css and javascript), that could mess with the old assets that came up with demo (I honestly have no idea if this is either possible or not).
- My researching results hint that what caused the problem might be the way the CSS files are included (Ex: missing the property type="text/css" for link tag), but I don't want to change anything in the code base because I am not 100% convinced I understand it.
- It might be just the environment in which I host the app (I used namecheap hosting).

I WOULD APPRECIATE ANY THOUGHTS  PUT ONTO THIS TO HELP ME FIND A SOLUTION, BECAUSE I DON'T WANT TO START OVER AGAIN BUILDING AN ADMIN DASHBOARD FROM SCRATCH.
MANY THANKS IN ADVANCE.

![Screenshot from 2023-07-15 12-03-29](https://github.com/vanilophp/framework/assets/118732091/340c2907-fda2-4efb-a6f8-2c4ce2fc1392)
![Screenshot from 2023-07-15 12-02-38](https://github.com/vanilophp/framework/assets/118732091/fd992093-6e41-4e39-a478-e71fd3840c3a)
![Screenshot from 2023-07-15 11-58-23](https://github.com/vanilophp/framework/assets/118732091/66535fcf-42c6-4497-b784-827a4945789c)

**Comment by fulopattila122** [comment]
Is there an `appshell.css` file in your app's public/css folder?

**Reply by kassihassan** [reply]
yes.
![Screenshot from 2023-07-17 09-22-54](https://github.com/vanilophp/framework/assets/118732091/0249734d-075e-490e-b5ab-0220dd100c4e)

**Comment by fulopattila122** [comment]
Does `/css/app.css` load properly?

**Reply by kassihassan** [reply]
I don't think the website loads /css/app.css in production mode, I think it loads /build/assets/app-81db6e91.css instead,
![hashed-css](https://github.com/vanilophp/framework/assets/118732091/bd43add3-e519-4663-9469-8e2a46aa9982)

**Comment by fulopattila122** [comment]
This also gives 404: https://software.kassihassan.com/css/app.css

This must be a web server configuration issue.

Although generic web development support is out of my scope here, if you let me know 1) how your web server is configured and, 2) what is your document root, 3) where is the `build/` folder; I might be able to give some hints.

**Reply by kassihassan** [reply]
Thanks anyway.
1. I am not sure what configuration we're talking about here
2. I hosted the whole website as subdomain in a different folder (not /public_html), but I don't think that's the problem.
3. the two folders have the same root, public directory (public/build/assets & public/css)

**Comment by fulopattila122** [comment]
Oh, I see you're using LiteSpeed. Unfortunately, my knowledge of LiteSpeed is almost zero.

![image](https://github.com/vanilophp/framework/assets/1162360/3e081c3b-92c4-472a-bd42-c6ec45bd0d36)

**Reply by kassihassan** [reply]
It's okay, all I need is just understanding what happened and what should I do next. (I think appshell.css should be moved somehow to build/assets and served just like other css files, but I have no idea how to do it without changing some code in vendor folder)

**Comment by fulopattila122** [comment]
Unfortunately, I can't help with this. The web server should load those files from the `public/css` folder.

As an alternative, you can move them to the build folder (because that works) and tell the admin panel to load them from a different location: https://konekt.dev/appshell/3.x/admin-panel#custom-assets

**Reply by kassihassan** [reply]
I'll try, thank you for your guidance.

**Reply by fulopattila122** [reply]
Please let us know how it went and what the solution was once it works. Thank you :bow:

**Reply by kassihassan** [reply]
I will for sure.
